### ISSUE TITLE
Description of filter function is wrong

### ISSUE THREAD

**Issue by gspringtech** [Lambda-X/cljs-repl-web#225]
> Description
    > Returns a lazy sequence of the non-nil results of (f item). 
    > Note, this means false return values will be included.

The first sentence is dubious, and the second sentence is wrong. (Unless I'm missing some very strange behaviour in ClojureScript.)
Contrast this to clojure.core:
    
    > Returns a lazy sequence of the items in coll for which (pred item) returns true.

**Comment by arichiardi** [comment]
You are right, it definitely resembles `keep. Thanks for reporting and PR is welcome!
